src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -320,6 +320,62 @@ public List<JobModel> enumerateJobs(
         .collect(Collectors.toList());
   }
 
+  private List<FlightState> getAccessibleFlights(
+      int offset, int limit, FlightFilter filter, AuthenticatedUserRequest userReq)
+      throws InterruptedException {
+    int start = 0;
+    int end = offset + limit;
+    HashMap<String, List<String>> roleMap = new HashMap<>();
+    ArrayList<FlightState> flightStateList = new ArrayList<>();
+    while (flightStateList.size() < offset + limit) {
+      List<FlightState> filterResults = stairway.getFlights(start, end, filter);
+      if (filterResults.size() == 0) {
+        break;
+      }
+      filterResults.forEach(
+          flightState -> {
+            if (userLaunchedFlight(flightState, userReq)) {
+              flightStateList.add(flightState);
+            } else {
+              FlightMap inputParameters = flightState.getInputParameters();
+              IamResourceType resourceType =
+                  inputParameters.get(
+                      JobMapKeys.IAM_RESOURCE_TYPE.getKeyName(), IamResourceType.class);
+              String resourceId =
+                  inputParameters.get(JobMapKeys.IAM_RESOURCE_ID.getKeyName(), String.class);
+              IamAction action =
+                  inputParameters.get(JobMapKeys.IAM_ACTION.getKeyName(), IamAction.class);
+              if (resourceType != null & resourceId != null && action != null) {
+                String key = String.format("%s-%s", resourceType, resourceId);
+                List<String> userRoles = roleMap.get(key);
+                if (userRoles == null) {
+                  userRoles = samService.listActions(userReq, resourceType, resourceId);
```

Comment:
That's a great suggestion!

Reply:
Hmm for some reason I only see resource roles in this endpoint 🤔. Here's an example:
```
{
    "authDomainGroups": [],
    "direct": {
      "actions": [],
      "roles": [
        "steward"
      ]
    },
    "inherited": {
      "actions": [],
      "roles": []
    },
    "missingAuthDomainGroups": [],
    "public": {
      "actions": [],
      "roles": []
    },
    "resourceId": "d90d0c95-ba7a-4874-bed3-9dfc219a01ab"
  }```

Reply:
I just asked for clarification from the Identiteam!

Reply:
Ah rats, I can replicate this on my end!

I see you posted in #dsp-identiteam, good move :)

Reply:
I also don't consider this a blocker to merge, by the way.  If it ultimately is a bug on Identiteam's end that they fix, I'd guess it would take awhile for that fix to propagate.  If we were inclined, we could do a small refactor in a follow-on PR to reduce the SAM calls.